src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/CreateCalendarEvent/CreateCalendarEventCommandValidator.cs

```diff
@@ -0,0 +1,246 @@
+                });
+
+                When(c => c.EventFormat == EventFormat.Online, () =>
+                {
+                    RuleFor(c => c.Location)
+                        .Empty()
+                        .WithMessage(LocationMustBeEmpty);
+
+                    RuleFor(c => c.Postcode)
+                        .Empty()
+                        .WithMessage(PostcodeMustBeEmpty);
+
+                    RuleFor(c => c.Latitude)
+                        .Empty()
+                        .WithMessage(LatitudeMustBeEmpty);
+
+                    RuleFor(c => c.Longitude)
+                        .Empty()
+                        .WithMessage(LongitudeMustBeEmpty);
+
+                    RuleFor(c => c.EventLink)
+                        .MaximumLength(2000)
+                        .WithMessage(EventLinkMustNotExceedLength)
+                        .Matches(Constants.RegularExpressions.UrlRegex)
+                        .WithMessage(EventLinkMustBeValid);
+                });
+
+                When(c => c.EventFormat == EventFormat.Hybrid, () =>
+                {
+                    RuleFor(c => c.Location)
+                        .NotEmpty()
+                        .WithMessage(LocationMustNotBeEmpty)
+                        .MaximumLength(200)
+                        .WithMessage(LocationMustNotExceedLength);
+
+                    RuleFor(c => c.Postcode)
+                        .NotEmpty()
+                        .WithMessage(PostcodeMustNotBeEmpty)
+                        .Matches(Constants.RegularExpressions.PostcodeRegex)
+                        .WithMessage(PostcodeMustBeValid);
+
+                    RuleFor(c => c.Latitude)
+                        .NotEmpty()
+                        .WithMessage(LatitudeMustNotBeEmpty)
+                        .InclusiveBetween(-90, 90)
+                        .WithMessage(LatitudeMustBeValid);
+
+                    RuleFor(c => c.Longitude)
+                        .NotEmpty()
+                        .WithMessage(LongitudeMustNotBeEmpty)
+                        .InclusiveBetween(-180, 180)
+                        .WithMessage(LongitudeMustBeValid);
+
+                    RuleFor(c => c.EventLink)
+                        .MaximumLength(2000)
+                        .WithMessage(EventLinkMustNotExceedLength)
+                        .Matches(Constants.RegularExpressions.UrlRegex)
+                        .WithMessage(EventLinkMustBeValid);
+                });
+
+                RuleFor(c => c.ContactName)
+                    .NotEmpty()
+                    .WithMessage(ContactNameMustNotBeEmpty)
+                    .MaximumLength(200)
+                    .WithMessage(ContactNameMustNotExceedLength);
+
+                RuleFor(c => c.ContactName)
+                    .NotEmpty()
+                    .WithMessage(ContactNameMustNotBeEmpty)
+                    .MaximumLength(200)
+                    .WithMessage(ContactNameMustNotExceedLength);
```

Comment:
duplicate rule around contactname (Remove this one)

Reply:
Good spot